docs/API/core/app-actions.md

```diff
@@ -326,6 +326,42 @@ Fliplet.App.Actions.remove('confirm-booking').then(function () {
 
 ---
 
+## Testing and debugging an action
+
+Once you have created an app action and added the code you want to run to the onRemoteExecution() function you should test it and see if it running as expected. Note that you cannot use developer tools to view your console.log inside the onRemoteExecution() function as this is running on the server side. Here is the general steps to test an app action: 
+
+1. Create the app action 
+2. Add code to onRemoteExecution() that you want to run in the app action. 
+3. Call the app action on demand and log the returned data to see if it working as expected. 
+4. If there are multiple steps in your code then move the `return Promise.resolve()` to the point where you want to see the data available. 
+
+```js
+// If an app action called 'test-action' exists then we can call in on another screen
+Fliplet.App.Actions.runWithResult('test-action').then(function (result) {
+  // If our app action working then it should return the entries from the data source query
+  console.log(result)
+});
+
+//This code exists on another screen and queries the data source and returns the data
+Fliplet.Page.onRemoteExecution(function (payload) {
+  return Fliplet.DataSources.connect(123)
+    .then(function (connection) {
+      return connection.find({
+        where: {
+            Name: 'John'
+        }
+      });
+    })
+    .then(function (records) {
+        //In order to see the data currently we need to return it in promise so that we can log in when the action is called above. 
+        return Promise.resolve(records)
+    });
+});
+
+```
+
+Note: You have to ensure that if you are executing asynchronous javaScript then the promises are chained correctly. If you do not chain the promises correctly then it is likely that no payload will be returned. 
```

Comment:
"JavaScript".

This however applies to all our JS API, not just App Actions so I don't think it should be here. @tonytlwu thoughts?

Reply:
I agree but I have found that app actions are a lot less forgiving and if you chain incorrectly it just will not run properly. It's a very common mistake and I just want to highlight here for people so if they find something is not running correctly they review their promises.